src/ng/raf.js

```diff
@@ -24,8 +24,42 @@ function $$RAFProvider() { //rAF
           };
         };
 
-    raf.supported = rafSupported;
+    queueFn.supported = rafSupported;
 
-    return raf;
+    var cancelLastRAF;
+    var taskCount = 0;
+    var taskQueue = [];
+    return queueFn;
+
+    function flush() {
+      for (var i = 0; i < taskQueue.length; i++) {
+        var task = taskQueue[i];
+        if (task) {
+          taskQueue[i] = null;
+          taskCount--;
+          task();
+        }
+      }
+      taskCount = taskQueue.length = 0;
+    }
+
+    function queueFn(asyncFn) {
+      var index = taskQueue.length;
+
+      taskCount++;
+      taskQueue.push(asyncFn);
+
+      if (index === 0) {
+        cancelLastRAF = rafFn(flush);
+      }
+
+      return function() {
+        taskQueue[index] = null;
+        if (--taskCount === 0 && cancelLastRAF) {
+          cancelLastRAF();
+          cancelLastRAF = null;
+        }
+      };
+    }
```

Comment:
This function seems broken altogether.

What if it is called **after** the next raf ?
Won't it cancel another unrelated task ?

Also, cancelling will not properly empty the queue (it will leave it full of `null`s).